GitUI/ScriptsEngine/IRunScript.cs

```diff
@@ -4,5 +4,14 @@
     public interface IRunScript
     {
         void Execute(int scriptId);
+
+        void Execute(int scriptId, IFileListSource files);
+
+        public interface IFileListSource
```

Comment:
I can't see any reason to nest interfaces, it makes the API look rather unclean.

Reply:
The thing is, calling it simply `IFileListSource` would be an invitation for collision (since it only applies to scripts, and is not for generic file lists). Calling it `IScriptFileListSource` would be confusing, because it's returning files used by the script, not script files. And calling it `IScriptRunFileListSource` or `IFileListForScriptSource` would be incredibly verbose, and doesn't seem like an improvement over `IRunScript.IFileListSource`. But, if it must have a different (non-namespaced) name, and you have a suggestion, I'll change it.

Reply:
This is not idiomatic; .NET SDK does define any nested interfaces.